reconcile/skupper_network/reconciler.py

```diff
@@ -0,0 +1,269 @@
+import logging
+from collections.abc import Iterable
+from typing import Any
+
+from sretoolbox.utils import threaded
+
+import reconcile.openshift_base as ob
+from reconcile.skupper_network.models import SkupperSite
+from reconcile.skupper_network.site_controller import LABELS as SITE_CONTROLLER_LABELS
+from reconcile.skupper_network.site_controller import get_site_controller
+from reconcile.utils.oc import OC_Map
+from reconcile.utils.openshift_resource import OpenshiftResource as OR
+from reconcile.utils.openshift_resource import ResourceInventory
+
+
+def delete_skupper_site(
+    site: SkupperSite,
+    oc_map: OC_Map,
+    dry_run: bool,
+    integration_managed_kinds: Iterable[str],
+) -> None:
+    """Delete all skupper resources (leftovers not covered by ResourceInventory) starting with 'skupper-' in a namespace."""
+    oc = oc_map.get_cluster(site.cluster.name)
+    to_delete: dict[str, dict[str, Any]] = {}
+
+    for kind in integration_managed_kinds:
+        # delete everything labeled by us
+        to_delete.update(
+            {
+                f'{item["kind"]}-{item["metadata"]["name"]}': item
+                for item in oc.get_items(
+                    kind=kind,
+                    namespace=site.namespace.name,
+                    labels=SITE_CONTROLLER_LABELS,
+                )
+            }
+        )
+        # delete everything else that starts with 'skupper-'
+        to_delete.update(
+            {
+                f'{item["kind"]}-{item["metadata"]["name"]}': item
+                for item in oc.get_items(kind=kind, namespace=site.namespace.name)
+                if item["metadata"]["name"].startswith("skupper-")
+            }
+        )
+
+    for item in to_delete.values():
+        logging.info(
```

Comment:
should we also revoke the site tokens (outgoing to other sites) ?

Reply:
Obsolete site tokens are deleted by [delete_unused_tokens](https://github.com/app-sre/qontract-reconcile/blob/ae85b58f16cf564b3c0a1bef80e9f0550593d7d3/reconcile/skupper_network/reconciler.py#L208). The `delete_skupper_site` deletes everything else that is not managed by the integration, e.g., the internal certificates and keys created by the `site-controller`.